Engine/Entities/Colonies/Colony.cs

```diff
@@ -0,0 +1,132 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Numerics;
+using System.Reflection.Metadata.Ecma335;
+using AntEngine.Entities.States;
+using AntEngine.Entities.States.Living;
+using AntEngine.Maths;
+using AntEngine.Resources;
+
+namespace AntEngine.Entities.Colonies
+{
+    /// <summary>
+    /// Entity that can spawn other entities with resources.
+    /// </summary>
+    public class Colony : LivingEntity
+    {
+        private const string ColonyDefaultName = "Colony";
+
+        public delegate IColonyMember ColonySpawnMethod(string name, Transform transform, World world, Colony colony);
+        
+        private List<IColonyMember> _population;
+        private ResourceDeposit _stockpile;
+        private ResourceDeposit _spawnCost;
+
+        public Colony(World world, ColonySpawnMethod spawnMethod) : this(ColonyDefaultName, new Transform(), world, spawnMethod)
+        {
+        }
+
+        public Colony(string name, Transform transform, World world, ColonySpawnMethod spawnMethod) : base(name, transform, world, new IdleState())
+        {
+            _population = new List<IColonyMember>();
+            _stockpile = new ResourceDeposit();
+            _spawnCost = new ResourceDeposit();
+
+            SpawnMethod = spawnMethod;
+        }
+
+        /// <summary>
+        /// Current population of the colony.
+        /// This is basically the entities that have been spawned by the colony.
```

Comment:
```suggestion
        /// This corresponds to a list of the entities that have been spawned by the colony.
```

Basically sounds very "amateur". 
I'm being annoying, I know :/